codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/RuntimeClientPlugin.java

```diff
@@ -38,15 +39,39 @@
 public final class RuntimeClientPlugin implements ToSmithyBuilder<RuntimeClientPlugin> {
 
     private final Symbol resolveFunction;
+    private final ApplyPlugin buildMiddlewareStack;
     private final BiPredicate<Model, ServiceShape> servicePredicate;
     private final OperationPredicate operationPredicate;
 
     private RuntimeClientPlugin(Builder builder) {
         resolveFunction = builder.resolveFunction;
         operationPredicate = builder.operationPredicate;
         servicePredicate = builder.servicePredicate;
+        buildMiddlewareStack = builder.buildMiddlewareStack;
     }
 
+    @FunctionalInterface
+    public interface ApplyPlugin {
```

Comment:
Why not just take a symbol pointing to a function that does the registration? This seems to encourage placing quite a bit of what is usually static code into the code generator.

Reply:
Based on the teams discussion, this will be addressed in a seperate PR as an incremental task